cocos/core/effect-settings.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+  worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+  not use Cocos Creator software for developing other software or tools that's
+  used for developing games. You are not granted to publish, distribute,
+  sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+import { HTML5 } from 'internal:constants';
+import { legacyCC } from './global-exports';
+
+declare const fsUtils: any;
+
+/**
+ * @zh
+ * Settings 中的默认分组，通常与模块一一对应。
+ *
+ * @en
+ * The default grouping in Settings, which usually corresponds to the module.
+ */
+enum Category {
+    PATH = 'path',
+    ENGINE = 'engine',
+    ASSETS = 'assets',
+    SCRIPTING = 'scripting',
+    PHYSICS = 'physics',
+    RENDERING = 'rendering',
+    LAUNCH = 'launch',
+    SCREEN = 'screen',
+    SPLASH_SCREEN = 'splashScreen',
+    ANIMATION = 'animation',
+    PROFILING = 'profiling',
+    PLUGINS = 'plugins',
+}
+
+/**
+ * @zh
+ * 配置模块用于获取 settings.json 配置文件中的配置信息，同时你可以覆盖一些配置从而影响引擎的启动和运行，可参考 [game.init] 的参数选项说明。你可以通过 [settings] 访问此模块单例。
+ * @en
+ * The Settings module is used to get the configuration information in the settings.json configuration file,
+ * and you can override some of the configuration to affect the launch and running of the engine, as described in the [game.init] parameter options.
+ * You can access this single instance of the module via [settings].
+ */
+export class EffectSettings {
```

Comment:
What's the relationship between Settings and EffectSetting?

Reply:
EffectSetting has nothing to do with Settings. It is the serialization result of layout-graph and shader bindings, which are used in custom-pipeline.
EffectSettings cannot be made asset, because it is generated by the asset extension. As a result, we use settings to save/load it.